### DISCUSSION TITLE
[V3] form object update ... Error encoding model [xxx] with ID [1] to JSON: Recursion detected

### DISCUSSION THREAD

**Discussion by camiant** [livewire/livewire#6161] [1. Help]
According to docs 
[https://livewire.laravel.com/docs/forms#extracting-a-form-object](https://livewire.laravel.com/docs/forms#extracting-a-form-object)

```
public function setPost(Post $post)
{
    $this->post = $post;
 
    $this->title = $post->title;
 
    $this->content = $post->content;
}

public function store()
{
    Post::create($this->only(['title', 'content']));
}

public function update()
{
    $this->post->update(
        $this->all()
    );
}
```

While `store` method is ok, 'cause it takes only `title` and `content`, the array `$this->all()` contains the whole `$post` object too... so error `Error encoding model [...\Post] with ID [1] to JSON: Recursion detected` is thrown

Am I missing something ? 🤔

**Comment by joshhanley** [comment]
@camiant yeah that's correct, `all()` will include the post object. So using `only()` is better. PRs to update the docs are welcome! 🙂
